optimizely/config/polling_manager.go

```diff
@@ -53,42 +53,40 @@ type PollingProjectConfigManager struct {
 	exeCtx utils.ExecutionCtx // context used for execution control
 }
 
-func (cm *PollingProjectConfigManager) activate(initialPayload []byte, init bool) {
-
-	update := func() {
-		var e error
-		var code int
-		var payload []byte
-		if init && len(initialPayload) > 0 {
-			payload = initialPayload
-		} else {
-			payload, code, e = cm.requester.Get()
-
-			if e != nil {
-				cmLogger.Error(fmt.Sprintf("request returned with http code=%d", code), e)
-			}
-		}
-
-		projectConfig, err := datafileprojectconfig.NewDatafileProjectConfig(payload)
-		if err != nil {
-			cmLogger.Error("failed to create project config", err)
+func (cm *PollingProjectConfigManager) Sync(datafile []byte) {
+	var e error
+	var code int
+	if len(datafile) == 0 {
+		datafile, code, e = cm.requester.Get()
+
+		if e != nil {
+			cmLogger.Error(fmt.Sprintf("request returned with http code=%d", code), e)
 		}
+	}
 
-		cm.configLock.Lock()
-		cm.projectConfig = projectConfig
-		cm.err = err
-		cm.configLock.Unlock()
+	projectConfig, err := datafileprojectconfig.NewDatafileProjectConfig(datafile)
+	if err != nil {
+		cmLogger.Error("failed to create project config", err)
 	}
 
+	cm.configLock.Lock()
+	cm.projectConfig = projectConfig
+	cm.err = err
+	cm.configLock.Unlock()
+}
+
+func (cm *PollingProjectConfigManager) start(initialDatafile []byte, init bool) {
```

Comment:
Wasn't sure `activate` was the right method name.

Reply:
does not matter, activate activates polling manager, I have no problem with that name. but you can use start too, it does not matter